start.sh

```diff
@@ -100,7 +100,18 @@ if [ -n "${PIHOLE_DNS_}" ]; then
     PIHOLE_DNS_ARR=(${PIHOLE_DNS_//;/ })
     count=1
     for i in "${PIHOLE_DNS_ARR[@]}"; do
-        change_setting "PIHOLE_DNS_$count" "$i"
+        # Split upstream address by #
+        split=(${i//#/ })
+
+        # Resolve hostname if any, otherwise use ip
+        resolved="$(ping -q -c 1 -t 1 "$split" | grep PING | sed -e "s/).*//" | sed -e "s/.*(//")"
```

Comment:
Should we also filter to only do this if non-digits are found (e.g. this is actually a hostname)? Otherwise we're wasting time and networking to ping/dig something we don't need to. With ping, in particular, there's now a requirement that the system accepts pings and is up at launch, which wasn't a requirement before to start up.

Reply:
I think this function is available in `start.sh`, as it [sources `webpage.sh`](https://github.com/pi-hole/docker-pi-hole/blob/master/start.sh#L44), which in turn [sources `basic-install.sh`](https://github.com/pi-hole/pi-hole/blob/master/advanced/Scripts/webpage.sh#L27)

https://github.com/pi-hole/pi-hole/blob/master/automated%20install/basic-install.sh#L1020-L1038